llvm/cmake/modules/HandleLLVMOptions.cmake

```diff
@@ -1285,7 +1285,7 @@ elseif(LLVM_ENABLE_LTO)
   endif()
 endif()
 
-if(LLVM_ENABLE_FATLTO AND UNIX AND NOT APPLE)
+if(LLVM_ENABLE_FATLTO AND ((UNIX AND NOT APPLE) OR FUCHSIA))
```

Comment:
I don't understand why that diff is required since you're setting it properly in your cache files.

Reply:
Because Fuchsia Targets aren't UNIX. Without this change `-ffat-lto-objects` won't be passed to `clang` through CMake. The additional logic is basically to restrict the flag to ELF targets, and it missed Fuchsia when we added it, since we weren't building our runtimes with FatLTO initially, just the toolchain + tests.

Reply:
Sorry, I think I initially read `if (LLVM_ENABLE_FATLTO OR ((UNIX AND NOT APPLE) OR FUCHSIA))`.